packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -612,6 +613,16 @@ export class TreeCheckout implements ITreeCheckoutFork {
 
 				// TODO:#23442: When a revertible is cloned for a forked branch, optimize to create a fork of a revertible branch once per revision NOT once per revision per checkout.
 				const forkedCheckout = getCheckout(forkedBranch);
+
+				// Check if the original branch and forked branch have a common ancestor.
+				const originalBranchHead = this.#transaction.activeBranch.getHead();
+				const forkedBranchHead = forkedCheckout.#transaction.activeBranch.getHead();
+
+				assert(
+					findCommonAncestor(originalBranchHead, forkedBranchHead) !== undefined,
+					0x576 /* branch A and branch B must be related */,
+				);
```

Comment:
This should be done in a separate PR, and it should throw a UsageError instead of asserting.

Reply:
https://github.com/microsoft/FluidFramework/pull/23485

PR Created 😄 .